doc/compiler/osr/OSR.md

```diff
@@ -0,0 +1,276 @@
+                 /           |
+  ---------------   --------------------------
+  |call B(cold) |   |    call C (from B)     |---> catch 2
+  ---------------   --------------------------
+       |         \           |
+    catch 1       \          |
+                    --------------------------
+                    |                        |---> catch 1
+                    --------------------------
+                             |
+                    --------------------------    ---------------------    ---------------------
+                    |     exit               |    | catch 2           |    | catch 1           |
+                    --------------------------    ---------------------    ---------------------
+                                                          |                         |
+                                                  ---------------------    ---------------------
+                                                  |prepareForOSRHelper|___>|prepareForOSRHElper|___>  exit
+                                                  |   iconst 0        |    |  iconst -1        |
+                                                  ---------------------    ---------------------
+
+After OSRExceptionEdgeRemoval (removes all OSR exception edges except those that originate from OSR induce blocks):
+
+
+                    --------------------------
+                    |     entry              |
+                    --------------------------
+                             |
+                    --------------------------
+                    |   OSR guard            |
+                    --------------------------
+                   /         |
+                  /          |
+  ----------------  --------------------------
+  |jitInduceOSR  |  |  non-overriden guard 1 |
+  |AtCurrentPC   |  |                        |
+  ----------------  --------------------------
+        |          /         |
+    catch 1       /          |
+                 /           |
+  ---------------   --------------------------
+  |call B(cold) |   |    call C (from B)     |
+  ---------------   --------------------------
+                 \           |
+                  \          |
+                    --------------------------
+                    |                        |
+                    --------------------------
+                             |
+                    --------------------------    ---------------------
+                    |     exit               |    | catch 1           |
+                    --------------------------    ---------------------
+                                                          |
+                                                  ---------------------
+                                                  |prepareForOSRHelper|___>  exit
+                                                  |   iconst -1       |
+                                                  ---------------------
+
+
+```
```

Comment:
I would suggest making the CFGs code blocks and having the introductory lines, like "After ILGen:" in ordinary type.